Sources/SWBTaskExecution/DynamicTaskSpecs/CompilationCachingDataPruner.swift

```diff
@@ -140,6 +141,75 @@ package final class CompilationCachingDataPruner: Sendable {
         }
     }
 
+    package func pruneCAS(
+        _ toolchainCAS: ToolchainCAS,
+        key: ClangCachingPruneDataTaskKey,
+        activityReporter: any ActivityReporter,
+        fileSystem fs: any FSProxy
+    ) {
+        let casOpts = key.casOptions
+        guard casOpts.limitingStrategy != .discarded else {
+            return // No need to prune, CAS directory is getting deleted.
+        }
+        let inserted = state.withLock { $0.prunedCASes.insert(key).inserted }
+        guard inserted else {
+            return // already pruned
+        }
+
+        startedAction()
+        let serializer = MsgPackSerializer()
+        key.serialize(to: serializer)
+        let signatureCtx = MD5Context()
+        signatureCtx.add(string: "ClangCachingPruneData")
+        signatureCtx.add(bytes: serializer.byteString)
+        let signature = signatureCtx.signature
+
+        let casPath = casOpts.casPath.str
+        let path = key.path.str
+
+        // Avoiding the swift concurrency variant because it may lead to starvation when `waitForCompletion()`
```

Comment:
`waitForCompletion()` is async now, though

Reply:
I'm going to clean this up later alongside the other half of the pruning implementation